server/sonar-server/src/main/java/org/sonar/server/component/index/ComponentIndex.java

```diff
@@ -107,6 +107,7 @@ private static TopHitsBuilder createSubAggregation(ComponentIndexQuery query) {
   private QueryBuilder createQuery(ComponentIndexQuery query, ComponentTextSearchFeature... features) {
     BoolQueryBuilder esQuery = boolQuery();
     esQuery.filter(authorizationTypeSupport.createQueryFilter());
+    query.getComponentUuids().ifPresent(componentUuids -> esQuery.filter(termsQuery("_id", componentUuids)));
```

Comment:
`"_id"` should be moved to a constant (for example into `DefaultIndexSettings`) - but this does not have to be done in this PR.